restapi/src/main/java/io/stargate/web/docsapi/service/util/DocsApiUtils.java

```diff
@@ -61,6 +62,37 @@ public static String convertArrayPath(String path) {
     }
   }
 
+  /**
+   * Converts any literal unicode points into their actual character. E.g. if the input string is
+   * literally abc\u002E123, this function returns abc.123 This allows a user to use unicode escape
+   * sequence in where filters when it would otherwise be ambiguous to use the corresponding
+   * character. Any path that already has square brackets in it will be left alone.
+   *
+   * @param path single filter or field path
+   * @return Converted to a string with no literal unicode code points.
+   */
+  public static String convertUnicodeCodePoints(String path) {
```

Comment:
Minor naming suggestion: this relates to escaping more than Unicode code point handling (although escapes are used to encode code points of course) so maybe naming should be something indicating that here we are decoding possibly escaped unicode characters? I mean, indicate direction in name itself, in addition to Javadocs.